fhir-server/src/main/java/com/ibm/fhir/server/filter/rest/FHIRRestServletFilter.java

```diff
@@ -220,39 +218,43 @@ private void checkValidTenantConfiguration(String tenantId) throws Exception {
     }
 
     /**
-     * @return a map of HTTP request headers, keyed by header name
+     * Extracts a map of HTTP request headers, keyed by header name, and sets them in the request context.
      */
-    private Map<String, List<String>> extractRequestHeaders(HttpServletRequest request) {
+    private void extractRequestHeaders(FHIRRequestContext context, HttpServletRequest request) {
         // Uses LinkedHashMap just to preserve the order.
         Map<String, List<String>> requestHeaders = new LinkedHashMap<>();
-
         List<String> headerNames = Collections.list(request.getHeaderNames());
         for (String headerName : headerNames) {
             requestHeaders.put(headerName, Collections.list(request.getHeaders(headerName)));
         }
-
-        return requestHeaders;
+        context.setHttpHeaders(requestHeaders);
     }
 
-    private HTTPHandlingPreference computeHandlingPref(ServletRequest request) throws FHIRException {
-        HTTPHandlingPreference handlingPref = HTTPHandlingPreference.from(FHIRConfigHelper.getStringProperty(FHIRConfiguration.PROPERTY_DEFAULT_HANDLING, "strict"));
+    /**
+     * Computes the handling preference from the Prefer header value.
+     * The default handling preference for all interactions is STRICT.
+     * @param context the context containing the HTTP headers
+     * @throws FHIRException
+     */
+    private void computeHandlingPref(FHIRRequestContext context) throws FHIRException {
+        HTTPHandlingPreference handlingPref = HTTPHandlingPreference.from(FHIRConfigHelper.getStringProperty(FHIRConfiguration.PROPERTY_DEFAULT_HANDLING, HTTPHandlingPreference.STRICT.value()));
         boolean allowClientHandlingPref = FHIRConfigHelper.getBooleanProperty(FHIRConfiguration.PROPERTY_ALLOW_CLIENT_HANDLING_PREF, true);
         if (allowClientHandlingPref) {
-            String handlingPrefString = ((HttpServletRequest) request).getHeader(preferHeaderName + ":" + preferHandlingHeaderSectionName);
```

Comment:
can we remove the special handling in our FHIRHttpServletRequestWrapper now that we have the processing logic here instead?

Reply:
Updated.